doc/source/usage.rst

```diff
@@ -0,0 +1,194 @@
+
+* Other attributes specified in the yaml file can include (for countries)
+  ISO2/3-codes, or the list of countries included in a macro-region (i.e.,
+  a continent or large region).
+
+Tag
+~~~
+
+To avoid repetition (and subsequent errors), any number of yaml files
+can be used as “tags” using a nested list of dictionaries.
+
+There must be only one top-level entry in any yaml file to be used as
+tag.
+
+The files defining the tags must have a name starting with ``tag_``.
+
+.. code:: yaml
+
+   - <Tag>:
+     - Some Key:
+         description: a short description of the key
+
+When importing the codelist, any occurrence of ``<Tag>`` in a variable
+name will be replaced by every element in the Tag dictionary. The
+``<Tag>`` will also be replaced in any of the variable attributes.
+
+
+Mapping
+~~~~~~~
+
+Model mappings, defined on a per-model basis serve three different purposes:
+
+1. Define a list of model native regions that are to be selected (and
+   usually uploaded) from an IAM result. This also serves as an implicit
+   exclusion list for model native regions, since only explicitly
+   mentioned regions are selected.
+
+2. Allow for renaming of model native regions.
+
+3. Define how model native regions should be aggregated to common
+   regions.
+
+This example illustrates how such a model mapping looks like:
+
+.. code:: yaml
+
+   model: model_a
+   native_regions:
+     - region_a: alternative_name_a
+     - region_b
+   common_regions:
+     - common_region_1:
+       - region_a
+       - region_b
+     - common_region_2:
+       - ...
+
+**Notes**
+
+-  The names of the three top level keywords **model**,
+   **native_regions** and **common_regions** are fixed.
+-  Required properties are **model** and **at least** either
+   **native_regions** or **common_regions**. **Both** are **allowed** as
+   well.
+-  **model** (str): specifies the model name for which the mapping
+   applies.
+-  **native_regions** (list): list of model native regions serves as
+   a selection as to which regions to keep.
+
+   -  In the above example *region_a* is to be renamed to
+      *alternative_name_a*. This is done by defining a key-value pair
+      of *model_native_name: new_name*.
+   -  *region_b* is selected but the name is not changed.
+   -  Assuming *model_a* also defines a third region *region_c*,
+      since it is not mentioned it will be **dropped** from the data.
+
+-  **common_regions** (list): list of common regions which will be
+   computed as aggregates. They are defined as list entries which
+   themselves have a list of constituent regions. These constituent
+   regions must be model native regions.
+
+   -  **Important to note** the names of the constituent regions
+      **must** refer to the **original** model native region names. In
+      the above example *region_a* and *region_b* and **not**
+      *alternative_name_a*.
+
+Guidelines and variable naming conventions
```

Comment:
Suggest to move this to an own page "Guidelines"

Reply:
Added a guidelines page